dali/c_api/c_api.cc

```diff
@@ -82,21 +86,33 @@ void SetExternalInputTensors(daliPipelineHandle *pipe_handle, const char *name,
 
 }  // namespace
 
+
+void daliInitialize() {
+  if (!dali_initialized) {
```

Comment:
Why not just `std::call_once` ?

Reply:
I went for static variable, cause in future, when we have error codes, it'll be good to check in the API functions, if dali's initialized

Reply:
You can have both.

Reply:
Good point. Will do both